src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +63,1061 @@ const ActorInit Dm_Char08_InitVars = {
+        DynaPolyActor_Init(&this->dyna, 3);
+        DynaPolyActor_LoadMesh(play, &this->dyna, &sTurtleGreatBayTempleCol);
+        this->dynapolyInitialized = true;
+    }
+
+    this->palmTree1 = Actor_Spawn(&play->actorCtx, play, ACTOR_OBJ_YASI, thisx->world.pos.x + -80.0f,
+                                  thisx->world.pos.y + 390.0f, thisx->world.pos.z, 0, 0, 0, 1);
+    this->palmTree2 = Actor_Spawn(&play->actorCtx, play, ACTOR_OBJ_YASI, thisx->world.pos.x + 68.0f,
+                                  thisx->world.pos.y + 368.0f, thisx->world.pos.z - 174.0f, 0, 0x7530, 0, 1);
+
+    switch (play->sceneNum) {
+        case SCENE_31MISAKI:
+            if (gSaveContext.save.weekEventReg[53] & 0x20) {
+                thisx->world.pos.x = -6480.0f;
+                this->targetYPos = -120.0f;
+                thisx->world.pos.z = 1750.0f;
+                thisx->world.rot.x = 0;
+                thisx->world.rot.y = -0x7234;
+                thisx->world.rot.z = 0;
+                thisx->shape.rot.x = 0;
+                thisx->shape.rot.y = -0x7234;
+                thisx->shape.rot.z = 0;
+                this->unk_1F0 = 1.0f;
+                this->unk_1FF = 2;
+                this->animIndex = TURTLE_ANIM_FLOAT;
+                this->unk_203 = 0x63;
+                this->eyeMode = TURTLE_EYEMODE_1;
+                this->unk_207 = 0;
+                this->unk_208 = 0;
+                thisx->flags |= ACTOR_FLAG_1;
+                if (gSaveContext.save.entrance == ENTRANCE(ZORA_CAPE, 8)) {
+                    this->eyeMode = TURTLE_EYEMODE_0;
+                    this->actionFunc = func_80AAFAC4;
+                } else {
+                    this->actionFunc = func_80AAF8F4;
+                }
+            } else {
+                this->actionFunc = DmChar08_WaitForSong;
+            }
+            break;
+
+        case SCENE_SEA:
+            this->unk_1FF = 2;
+            this->animIndex = TURTLE_ANIM_FLOAT;
+            this->unk_203 = 0x63;
+            this->eyeMode = TURTLE_EYEMODE_0;
+            this->unk_207 = 0;
+            this->unk_208 = 0;
+            thisx->flags |= ACTOR_FLAG_1;
+            this->actionFunc = func_80AAFAE4;
+            this->unk_1F0 = 1.0f;
+            break;
+
+        case SCENE_KONPEKI_ENT:
+            this->unk_1FF = 2;
+            this->animIndex = TURTLE_ANIM_FLOAT;
+            this->unk_203 = 0x63;
+            this->eyeMode = TURTLE_EYEMODE_0;
+            this->unk_207 = 0;
+            this->unk_208 = 0;
+            this->actionFunc = DmChar08_DoNothing;
+            this->unk_1F0 = 1.0f;
+            break;
+    }
+    DmChar08_ChangeAnim(&this->skelAnime, &sAnimationInfo[this->animIndex], 0);
+}
+
+void DmChar08_Destroy(Actor* thisx, PlayState* play) {
+    DmChar08* this = THIS;
+
+    if (this->dynapolyInitialized) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+s16 sBigTurtleOcarinaSoundLatch = false;
```

Comment:
If I remember correctly, there was some debate around this name in the original PR. I suggested `sSuccessSoundAlreadyPlayed` at the time; what do you think?